sourcegraph/sourcegraph
# Code Insights: Remove dashboard query from class-based API

Follow up for https://github.com/sourcegraph/sourcegraph/pull/43778

## Background
In this PR, I remove dashboard-based query methods from the class-based API. Motivation for this change you can find in [this doc](https://docs.google.com/document/d/1GYLfV6VCB42YyoqWCoM9ACFaEnPQz1n5_veArEVCus4/edit#) in a nutshell - there is no longer needed to have API query/mutation in the class API. 

## Test plan
- Check that the dashboard page works as it worked before 
- Check that if you came from the dashboard page to the creation UI page - newly created insight should be included in the dashboard by default 
- Check that the edit UI for dashboards works as expected (you can change the title, and visibility fields and see it on the dashboard page)



## App preview:

- [Web](https://sg-web-vk-clean-up-api-backend-2.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
